drivers/ata/ahciem.c

```diff
@@ -276,10 +276,6 @@ static unsigned int ahciem_sesop_rxdx_2(struct ahciem_args *args, u8 *rbuf)
 		memcpy(rbuf + offset, args->enc->status[i], 4);
 
 		ap = host->ports[i];
-		if (!ap) {
-			rbuf[offset] |= ENCLOSURE_STATUS_NOT_INSTALLED;
-			continue;
-		}
```

Comment:
I think it would be nice to still differentiate disabled ports, even if not required.